compute/cker/include/cker/train/operation/BinaryArithmetic.h

```diff
@@ -51,6 +51,15 @@ void BinaryArithmeticGrad(const Shape &incoming_shape, const T *incoming_data,
     break;
 
     case ArithmeticType::kSub:
+    {
+      BroadcastTo(incoming_shape, const_cast<T *>(incoming_data), lhs_grad_shape, lhs_grad_data);
+
+      auto const in_map = MapAsMatrixWithLastDimAsRows(incoming_data, incoming_shape);
+      auto rhs_grad_map = MapAsMatrixWithLastDimAsRows(rhs_grad_data, rhs_grad_shape);
+      rhs_grad_map = -in_map;
```

Comment:
Thinking about `sub` gradient, I thought that for RHS multiplying `-1` is enough.

But here, Functions like `MapAsMatrixWithLastDimAsRows` is called.
If you don't mind, could you explain a bit - why this function is called? (I'm not familiar with Eigen :sweat_smile: )

Reply:
> Thinking about `sub` gradient, I thought that for RHS multiplying `-1` is enough.
> 
> But here, Functions like `MapAsMatrixWithLastDimAsRows` is called. If you don't mind, could you explain a bit - why this function is called? (I'm not familiar with Eigen 😅 )

To my understanding, `MapAsMatrixWithLastDimAsRows` is generates a multidimensional view of underlying data, and calling it itself does not modify anything. I thought that Implementing this way makes it clear that we're doing elementwise negation on the multidimensional data.

Reply:
Oh I see.  

(suggestion) 

I read the code and It looks like `MapAsMatrixWithLastDimAsRows` creates a data view for easy access alongside the last dimensions. If I understand correctly, it looks like there is no access alongside specific dimensions. 

If so, How about using `MapAsVector` instead of `MapAsMatrixWithDimAsRows`? 
Maybe like this..? 
```cpp 
      const auto incoming_map =  MapAsVector(incoming_data, incoming_shape);
      auto rhs_grad_map = MapAsVector(rhs_grad_data, rhs_grad_shape);
      rhs_grad_map = -1 * incoming_map.array() ; 
```

Reply:
Updated.